packages/core-state/src/dpos/dpos.ts

```diff
@@ -3,10 +3,11 @@ import { Utils } from "@arkecosystem/crypto";
 
 @Container.injectable()
 export class DposState implements Contracts.State.DposState {
-    @Container.inject(Container.Identifiers.Application)
-    private readonly app!: Contracts.Kernel.Application;
+    @Container.inject(Container.Identifiers.LogService)
+    private logger!: Contracts.Kernel.Logger;
 
     @Container.inject(Container.Identifiers.WalletRepository)
+    @Container.tagged("state", "blockchain") // TODO: see todo in block-state
```

Comment:
@rainydio Is this class designed to be used with a specific Container.Identifiers.WalletRepository (as per the change here), or is it intended to work with multiple tagged versions? See my comment above about the same issue.